runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -725,13 +788,53 @@ TR_RelocationRuntime::validateAOTHeader(TR_FrontEnd *fe, J9VMThread *curThread)
    return false;
    }
 
+#if defined(JITSERVER_SUPPORT)
+void *
+TR_RelocationRuntime::isROMClassInSharedCaches(UDATA romClassValue)
+   {
+   TR_ASSERT(0, "Error: isROMClassInSharedCaches not supported in this relocation runtime");
```

Comment:
Why don't we support them?
If we know that it is never going to be called for a pure TR_RelocationRuntime, then maybe we need to add these functions to the derived classes only.

Reply:
`isROMClassInSharedCaches ` is only implemented by `TR_SharedCacheRelocationRuntime`.  If it's not defined as `virtual`, `compInfo->reloRuntime()` needs to be casted as `TR_SharedCacheRelocationRuntime`. Not a lot of places call `isROMClassInSharedCaches `. Will make this change.

Reply:
I am probably ok with the existing implementation. It seems they can only be used at the client/non-jittas when AOT is on. At the server it's never going to work because there is a single RelocationRuntime stored in compInfo.
I would prefer those ASSERT to become FATAL_ASSERT. Since they are never executed (supposedly) there is no extra overhead.

Reply:
I agree. The base class (`TR_RelocationRuntime::is`) implementation is not called but after browsing the code, I realize casting them to `TR_SharedCacheRelocationRuntime` at where `reloRuntime()` is called might not be ideal.